### PR TITLE
Issue 110 lock tagging and other locking issues

### LINKED ISSUE #110: Lock tagging
Checkbox under permissions to allow teacher to prevent students from adding tags to posts

### PR DESCRIPTION
**Pull request by l3n0ire**
## Details

- Teacher can change the following permissions of the board in settings and when creating the board
    - allow commenting
    - allow liking
    - allow editing/creation/deletion of post
    - allow tagging

Closes #110, #67 , #107, #109

### CODE REVIEW

**Comment by Ammar-T** [comment]
@l3n0ire Do you mind moving all the locking likes/comments/adding... issues into one PR. I've noticed there's a lot of overlap in each PR so it's difficult to review. Maybe just keep the most recent locking PR because I think it'll include all the locking permissions.

**Review by JoelWiebe — APPROVED** [review]
This looks great.

Tested:
- tested each permission (e.g., tagging, moving, etc.) that it locks functionality for students
- teacher maintains full access to all permissions, even when turned off for students

I noticed that:
- students can create but not edit or delete a note (looks like a previous issue I will report as a separate bug
- the create new post permission requires the student to refresh before this permission takes effect (low priority feature I will report as a separate feature)

Great wording of all permission options. Glad to have this added functionality!

File: src/app/components/canvas/canvas.component.ts [hunk 1]
```diff
@@ -46,6 +47,8 @@ export class CanvasComponent {
   modeType = Mode
   fabricUtils: FabricUtils = new FabricUtils()
 
+  showAddPost:boolean
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.ts:50]:
```suggestion
  showAddPost: boolean = true
```

File: src/app/components/canvas/canvas.component.ts [hunk 2]
```diff
@@ -66,6 +70,8 @@ export class CanvasComponent {
     this.handleLikeButtonClick();
     this.postsService.observable(this.boardID, this.handleAddFromGroup, this.handleModificationFromGroup);
     this.boardService.observable(this.boardID, this.handleBoardChange);
+    this.showAddPost = true
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.ts:73]:
```suggestion
    
```

File: src/app/components/canvas/canvas.component.ts [hunk 3]
```diff
@@ -179,9 +186,14 @@ export class CanvasComponent {
...
-    this.boardService.update(this.boardID, { permissions: { allowStudentMoveAny: !value } })
-    this.lockPostsMovement(value)
+  updatePostPermissions = (permissions:Permissions) => {
+    this.boardService.update(this.boardID, { permissions:permissions})
+    this.lockPostsMovement(!permissions.allowStudentMoveAny)
+    this.updateShowAddPost(permissions)
+  }
+
+  updateShowAddPost(permissions:Permissions) {
+    this.showAddPost = (this.user.role =="student" && permissions.allowStudentEditAddDeletePost) || this.user.role =="teacher"
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.ts:196]: Just a bit cleaner.
```suggestion
    let isStudent = this.user.role == "student"
    let isTeacher = this.user.role == "teacher"
    this.showAddPost = (isStudent && permissions.allowStudentEditAddDeletePost) || isTeacher
```

File: src/app/components/canvas/canvas.component.ts [hunk 4]
```diff
@@ -296,7 +308,7 @@ export class CanvasComponent {
     this.canvas.on('mouse:down', e => {
       var post: any = e.target
       var likeButton = e.subTargets?.find(o => o.name == 'like')
-      if (likeButton) {
+      if (likeButton && ((this.user.role =="student" && this.board.permissions.allowStudentLiking) || this.user.role =="teacher") ) {
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.ts:311]:
```suggestion
      let isStudent = this.user.role =="student" 
      let isTeacher = this.user.role == "teacher"
      let studentHasPerm = isStudent && this.board.permissions.allowStudentLiking
      if (likeButton && (studentHasPerm || isTeacher) ) {
```

**Comment by JoelWiebe** [comment]
@StevenWuzz do you mind checking out this code when you get a chance and providing your review: accept/request for changes?

**Review by Ammar-T — APPROVED** [review]

**Pull request merged**